File: my-submissions/e1.py
# https://leetcode.com/problems/two-sum/

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        if len(nums) == 2 :
            return [0, 1]

        differences = {}
        for i in range(len(nums)) :
            if nums[i] in differences.keys() :
                return [i, differences.get(nums[i])]
            differences[target - nums[i]] = i
        return [0, 0]
        

File: my-submissions/e1.java
// https://leetcode.com/problems/two-sum/


class Solution {
    public int[] twoSum(int[] nums, int target) {
        if (nums.length == 2) {
            return new int[]{0, 1};
        }

        HashMap<Integer, Integer> ref = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            if (ref.containsKey(nums[i])) {
                return new int[]{ref.get(nums[i]), i};
            }

            ref.put((target - nums[i]), i);
        }

        // No solution (shouldn't occur given parameters provided)
        return new int[]{0, 0};
    }
}